### PR TITLE
Add behave tests

### PR DESCRIPTION
**Pull request by mec07**
Address issue #4

### CODE REVIEW

**Comment by mec07** [comment]
This needs to be rebased on top of master

**Comment by mec07** [comment]
Rebased on top of master

File: tests/pylato/test_hamiltonian.py [hunk 1]
```diff
@@ -70,31 +120,121 @@ def test_total_energy_happy_path(self):
...
+        # fake electrostatics (Hamilton.Wi)
+        Job.Hamilton.Wi = [1 for i in range(Job.NAtom)]
+
+        # Action
+        result = Job.Hamilton.add_H_collinear(Job, ii, jj)
+
+        # Result
+        assert result == expected_value
+
+    def test_buildH0_one_atom(self):
```
Comment (mec07) [inline tests/pylato/test_hamiltonian.py]: @horsfielda I think I just found a mistake in building H0. Before we were adding hopping terms from an atom onto itself, which I believe is incorrect. Right? I've corrected it above (I'll add a comment there too) and added this test to enforce this. Please let me know if I'm incorrect in thinking this though.

File: pylato/hamiltonian.py [hunk 2]
```diff
@@ -291,7 +279,7 @@ def buildH0(self, Job):
 
         # Step through all pairs of atoms
         for a1 in range(Job.NAtom):
-            for a2 in range(a1, Job.NAtom):
+            for a2 in range(a1+1, Job.NAtom):
```
Comment (mec07) [inline pylato/hamiltonian.py]: @horsfielda Here is the correction to stop an atom from hopping onto itself.

File: tests/pylato/test_hamiltonian.py (hunk 1 above)
Comment (mec07) [reply tests/pylato/test_hamiltonian.py]: Ahhh ooops... I've had a closer look at the code and we are explicitly expecting onsite blocks, which should be: `Job.Model.atomic[type]['e']`

File: pylato/hamiltonian.py (hunk 2 above)
Comment (mec07) [reply pylato/hamiltonian.py]: Turns out I was wrong. See comment below.

File: models/TBexp.json [hunk 3]
```diff
@@ -23,10 +23,12 @@
          "l": [0, 1], 
          "U": 10.0, 
          "I": 5.0,
-         "e": [[0.0, 0.0, 0.0, 0.0],
-                     [0.0, 1.0, 0.0, 0.0],
-                     [0.0, 0.0, 1.0, 0.0],
-                     [0.0, 0.0, 0.0, 1.0]],
+         "e": [
+             [0.0, 0.0, 0.0, 0.0],
```
Comment (mec07) [inline models/TBexp.json]: It looks like we're missing a diagonal value on this row... @mb4512 @horsfielda

Comment (mec07) [reply models/TBexp.json]: I've just added it in.

File: features/groundstate_energy.feature [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Feature: Find the groundstate energy
```
Comment (mec07) [inline features/groundstate_energy.feature]: @horsfielda I think you'll find this interesting. I've only actually added one behave (or black box) test. In the long term we would really like to have a ton of these, to really test PyLATO properly. But for now it's a really good way to ensure that any changes we make don't inadvertently stop PyLATO from running at all (like I broke it when I merged in the python3 branch!).

**Comment by horsfielda** [comment]
Hi Marc,

Thanks for the comment. This does sound interesting.

Andrew

**Pull request merged**